src/components/products/Products.jsx

```diff
@@ -45,14 +64,41 @@ const Products = ({ data }) => {
     }
   };
 
+  const filtering = (products) => {
+    let filteredProducts = [];
+    if (activeCategories.length) {
+      filteredProducts = products.filter((product) =>
+        activeCategories.includes(product.category)
+      );
+    } else {
+      filteredProducts = [...products];
+    }
+
+    if (activeSizes.length) {
+      filteredProducts = filteredProducts.filter((product) =>
+        activeSizes.includes(product.size)
+      );
+    }
+
+    filteredProducts = filteredProducts.filter((product) => {
+      return product.price < priceRange[1] && product.price > priceRange[0];
+    });
+
+    return filteredProducts;
+  };
+
+  useEffect(() => {
+    setProducts(filtering(data));
+  }, [activeCategories, activeSizes, priceRange]);
+
   useEffect(() => {
     setProducts(sortProducts(sorting));
-    setCurrentPage(1);
+    setCurrentPage(currentPage);
```

Comment:
При изменении порядка сортировки, мы хотим сбросить страницу на первую. Тогда можем сразу писать следующим образом
```suggestion
    dispatch(updateCurrentPage(1));
```
В идеале сделать выше `const FIRST_PAGE_NUMBER = 1;` и использовать эту константу